pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -169,6 +173,31 @@ def _run_early_stopping_check(self, trainer):
             if self.wait_count >= self.patience:
                 self.stopped_epoch = trainer.current_epoch
                 trainer.should_stop = True
+                self._log_info(
+                    f"Monitored metric {self.monitor} did not improve in the last {self.wait_count} epochs."
+                    f" Best score: {self.best_score:.3f}. Signaling Trainer to stop.",
+                    rank=trainer.global_rank,
+                )
 
         # stop every ddp process if any world process decides to stop
         trainer.should_stop = trainer.training_type_plugin.reduce_boolean_decision(trainer.should_stop)
+        if trainer.should_stop:
+            self._log_debug("Signaling Trainer to stop.", rank=trainer.global_rank)
+
+    def _improvement_message(self, current: torch.Tensor) -> str:
+        """ Formats a log message that informs the user about an improvement in the monitored score. """
+        if torch.isfinite(self.best_score):
+            msg = (
+                f"Metric {self.monitor} improved by {abs(self.best_score - current):.3f} and is exceeding"
+                f" min_delta = {abs(self.min_delta)}. New best score: {current:.3f}"
+            )
+        else:
+            msg = f"Metric {self.monitor} improved. New best score: {current:.3f}"
+        return msg
+
+    def _log_info(self, msg: str, rank: int) -> None:
```

Comment:
@awaelchli What's the point of having the `rank` parameter? Maybe `rank_zero_only`=True parameter would make more sense?

Reply:
it depends. we can log_info on each rank when verbose, like the way I have done it for debug a few lines below.
basically I implemented it like this so people can give feedback what they prefer.

Reply:
I think it would make sense to display on all ranks only if the `current_value` is different.

Reply:
it's different when the user does not use `sync_dist=True` in logging, but we don't know that.
The best I could do is `all_gather` and then compare. Is it worth it?

Reply:
Might be an over-kill. I don't have strong opinion on either `log_info` or `rank_zero_info`.